resources/packs/googleworkspace/connected_apps.go

```diff
@@ -0,0 +1,113 @@
+package googleworkspace
+
+import (
+	"go.mondoo.com/cnquery/resources/packs/core"
+	"go.mondoo.com/cnquery/stringx"
+)
+
+type connectedApp struct {
+	clientID string
+	scopes   []string
+	name     string
+	appType  string
+	users    []*mqlGoogleworkspaceUser
+	tokens   []*mqlGoogleworkspaceToken
+}
+
+func (g *mqlGoogleworkspace) GetConnectedApps() ([]interface{}, error) {
+	// get all users
+	users, err := g.Users()
+	if err != nil {
+		return nil, err
+	}
+
+	connectedApps := map[string]*connectedApp{}
+	for _, user := range users {
+		usr := user.(*mqlGoogleworkspaceUser)
+		// get all token from user
+		tokens, err := usr.GetTokens()
+		if err != nil {
+			return nil, err
+		}
+
+		for _, token := range tokens {
+			tk := token.(*mqlGoogleworkspaceToken)
+
+			clientID, err := tk.ClientId()
+			if err != nil {
+				return nil, err
+			}
+
+			cApp, ok := connectedApps[clientID]
+			if !ok {
+				cApp = &connectedApp{
+					clientID: clientID,
+					users:    []*mqlGoogleworkspaceUser{},
+					tokens:   []*mqlGoogleworkspaceToken{},
+				}
+			}
+
+			// assign name
+			displayText, err := tk.DisplayText()
+			if err != nil {
+				return nil, err
+			}
+			cApp.name = displayText
+
+			// merge scopes
+			scopes, err := tk.Scopes()
+			if err != nil {
+				return nil, err
+			}
+			stringScopes := []string{}
+			for _, scope := range scopes {
+				stringScopes = append(stringScopes, scope.(string))
+			}
+			cApp.scopes = stringx.DedupStringArray(append(cApp.scopes, stringScopes...))
+
+			cApp.tokens = append(cApp.tokens, tk)
+			cApp.users = append(cApp.users, usr)
+
+			connectedApps[clientID] = cApp
+		}
+	}
+
+	// group token by client id
+	runtime := g.MotorRuntime
+	res := []interface{}{}
+	for _, connectedApp := range connectedApps {
+		users := []interface{}{}
```

Comment:
nit: you could extract the conversion logic to a generic function, so we can easily re-use it. I know we had a function that converts a slice of strings to slice of interface. Now with generics we could have 1 function that works for all types. Should be something like this:
```go
func SliceToInterfaceSlice[T any](ss []T) []interface{} {
  ssI := make([]interface{}, 0, len(ss))
  for _, s := range ss {
    ssI = append(ssI, s)
  }
  return ssI
}
```

Reply:
Would be a great addition to core. Can you take this?

Reply:
I can do that. Should I do it in this PR or add it in another PR once this is merged?

Reply:
Lets do this as follow-up